forward_effort_controller doesn't work for URsim

**Software Version**
ROS2 humble; URsim polyscope 5.24.0

**Question**

Description: I am attempting to use the direct torque interface ( forward_effort_controller) with URSim. However, when I try to activate the controller, it fails to start.

The error log indicates that the effort command interfaces for the joints do not exist. I have confirmed this by checking the hardware interfaces list, which shows position and velocity interfaces are available, but effort is missing.

Steps to Reproduce

Launch the UR ROS2 driver connected to URSim.

Attempt to spawn/activate the forward_effort_controller.

Actual Behavior The controller manager rejects the switch with the following error:

[ur_ros2_control_node-1] [ERROR] [1768607649.764884155] [resource_manager]: Not acceptable command interfaces combination: 
[ur_ros2_control_node-1] Start interfaces: 
[ur_ros2_control_node-1] [
[ur_ros2_control_node-1]    shoulder_pan_joint/effort
[ur_ros2_control_node-1]    shoulder_lift_joint/effort
[ur_ros2_control_node-1]    elbow_joint/effort
[ur_ros2_control_node-1]    wrist_1_joint/effort
[ur_ros2_control_node-1]    wrist_2_joint/effort
[ur_ros2_control_node-1]    wrist_3_joint/effort
[ur_ros2_control_node-1] ]
...
[ur_ros2_control_node-1] Not existing: 
[ur_ros2_control_node-1] [
[ur_ros2_control_node-1]  shoulder_pan_joint/effort
...
[ur_ros2_control_node-1] ]
[ur_ros2_control_node-1] [ERROR] [1768607649.764955375] [controller_manager]: Could not switch controllers since prepare command mode switch was rejected.
Additional Context I ran ros2 control list_hardware_interfaces to inspect the available interfaces. As seen below, effort is missing from the command interfaces list (only position, velocity, and force_mode related interfaces are present):

command interfaces
    elbow_joint/position [available] [unclaimed]
    elbow_joint/velocity [available] [unclaimed]
    force_mode/damping [available] [unclaimed]
    ...
    shoulder_lift_joint/position [available] [unclaimed]
    shoulder_lift_joint/velocity [available] [unclaimed]
    shoulder_pan_joint/position [available] [unclaimed]
    shoulder_pan_joint/velocity [available] [unclaimed]
    ...
    wrist_1_joint/position [available] [unclaimed]
    wrist_1_joint/velocity [available] [unclaimed]
    ...
    (No effort interface listed)

I run the wrong version? or URsim is not supported for direct torque interface?

have you ever tested torque interface in URsim when you developed it?

thanks a lot!

--- a/ur_robot_driver/doc/usage/simulation.rst
+++ b/ur_robot_driver/doc/usage/simulation.rst
@@ -75,12 +75,13 @@ With this, we can spin up a driver using
 
 When we now move the robot in Polyscope, the robot's RViz visualization should move accordingly.
 
-For details on the (PolyScopr 5 and CB3) Docker image, please see the more detailed guide :ref:`here <ursim_docker>`.
+For details on the (PolyScope 5 and CB3) Docker image, please see the more detailed guide :ref:`here <ursim_docker>`.
 
 .. note::
    Although effort control is supported in PolyScope >= 5.23.0 / 10.10.0, it has no physical effect in URSim. 
    The controller manager will accept the effort interfaces, but the simulated robot will remain stationary.
 
+Mock hardware
 -------------
 
 The package can simulate hardware with the ros2_control ``MockSystem``. This emulator enables an

--- a/ur_robot_driver/doc/usage/simulation.rst
+++ b/ur_robot_driver/doc/usage/simulation.rst
@@ -78,7 +78,7 @@ When we now move the robot in Polyscope, the robot's RViz visualization should m
 For details on the (PolyScope 5 and CB3) Docker image, please see the more detailed guide :ref:`here <ursim_docker>`.
 
 .. note::
-   Although effort control is supported in PolyScope >= 5.23.0 / 10.10.0, it has no physical effect in URSim. 
+   Although effort control is supported in PolyScope >= 5.23.0 / 10.10.0, it has no effect in URSim. 
    The controller manager will accept the effort interfaces, but the simulated robot will remain stationary.
 
 Mock hardware

--- a/ur_robot_driver/doc/usage/simulation.rst
+++ b/ur_robot_driver/doc/usage/simulation.rst
@@ -77,7 +77,10 @@ When we now move the robot in Polyscope, the robot's RViz visualization should m
 
 For details on the (PolyScopr 5 and CB3) Docker image, please see the more detailed guide :ref:`here <ursim_docker>`.
 
-Mock hardware
+.. note::
+   Although effort control is supported in PolyScope >= 5.23.0 / 10.10.0, it has no physical effect in URSim. 
+   The controller manager will accept the effort interfaces, but the simulated robot will remain stationary.
+
 -------------
 
 The package can simulate hardware with the ros2_control ``MockSystem``. This emulator enables an

